sys/config/src/config_fcb.c

```diff
@@ -255,14 +262,84 @@ conf_fcb_append(struct conf_fcb *cf, char *buf, int len)
     if (rc) {
         return OS_EINVAL;
     }
-    fcb_append_finish(&cf->cf_fcb, &loc);
+    fcb_append_finish(fcb, &loc);
     return OS_OK;
 }
 
 static int
 conf_fcb_save(struct conf_store *cs, const char *name, const char *value)
 {
     struct conf_fcb *cf = (struct conf_fcb *)cs;
+
+    return conf_fcb_kv_save(&cf->cf_fcb, name, value);
+}
+
+void
+conf_fcb_compress(struct conf_fcb *cf,
+                  int (*copy_or_not)(const char *name, const char *val,
+                                     void *cn_arg),
+                  void *cn_arg)
+{
+    return conf_fcb_compress_internal(&cf->cf_fcb, copy_or_not, cn_arg);
+}
+
+static int
+conf_kv_load_cb(struct fcb_entry *loc, void *arg)
+{
+    struct conf_kv_load_cb_arg *cb_arg = arg;
+    char buf[CONF_MAX_NAME_LEN + CONF_MAX_VAL_LEN + 32];
```

Comment:
I see this `32` magic number quite a bit in this file.  Do you know what it represents?  I couldn't figure out why this extra buffer space is required (I know this was already here before your change).

Reply:
32 is probably carryover from older, happier times. I think '2' is now the correct magic number to add.

Reply:
Yep, seems like `2` is length of element data written to FCB here:
https://github.com/apache/mynewt-core/blob/master/fs/fcb/src/fcb_append.c#L106

However, since we read data by `data_off` and `data_len` I think there's no need to add any magic value here and just keep buffer large enough for element data only. I dug in code a bit and it seems this will work, but would prefer that someone more familiar with FCB confirm...

Reply:
FCB does not use extra space. config uses key=value'\0'. So therefore 2 extra. I don't think we check CONF_MAX_NAME_LEN nor CONF_MAX_VAL_LEN separately either; only together. Those numbers are there just to guide the user. sys/config tries to accomodate what it can.